### ISSUE TITLE
New Mutual Aid sea-map

### ISSUE THREAD

**Issue by ColmMassey** [DigitalCommons/mutual-aid-project#1]
We now need to build a multi source mutual aid sea-map

First new source is the new csv file here https://nextcloud.solidarityeconomy.coop/index.php/s/kQXnTxEDYFTmk8o

So we will need to build new graph for it first.

**Comment by ColmMassey** [comment]
Crossed wires here. When I say mutual-aid I am referring to the tiny Nextcloud file sourced data. I'll say covid-mutual-aid to refer to the big dataset.

**Comment by wu-lee** [comment]
> Ok. Casting any temporary hack aside, then, does it seem ok to spend the order of hours to a day decoupling the various `open-data` data sets

As discussed, I've started investigating this, seems feasible, and I've created a ticket to cover this work: https://github.com/SolidarityEconomyAssociation/technology-and-infrastructure/issues/61

**Comment by wu-lee** [comment]
@ColmMassey this is implemented, using ESSGLOBAL2.1, and merged into the`dev` branch, along with EG2.1 versions of covid-mutual-aid, ICA, and ICA Youth Network.

The one snag is that the data for this map is invalid, so it won't build currently. The reason is, the last line has no name, which is a mandatory field.  Suggested fix below (remove the `TRUE` value in the `approved` column), but we could also just remove it entirely, or add a name.  I don't think I have access, so perhaps you could do this @ColmMassey?

I need to investigate ways of getting feedback when the data is invalid like this.

```
--- original.csv	2021-05-01 17:57:56.391974620 +0100
+++ original.csv	2021-05-05 16:23:08.501754946 +0100
@@ -3,4 +3,4 @@
 2,Climate Cafe Hub,https://transitionliverpool.org/,,,53.38046;-2.9615,TRUE,sdfsdf
 3,Wirral Environmental Network,https://wirralenvironmentalnetwork.org.uk/,,,,TRUE,zxcxvcxvzxvxv
 4,Dale Farm Trust,https://www.dalefarmtrust.org/,,,53.41654;-3.02857,,
-5,,,,,53.33244;-3.11396,TRUE,xcvxvxvxcvxvxv
+5,,,,,53.33244;-3.11396,,xcvxvxvxcvxvxv
```

**Comment by ColmMassey** [comment]
I am confused about which spreadsheet we are using.

For some reason I have been populating this one...

https://nextcloud.solidarityeconomy.coop/index.php/s/YDctt5nbP4e7Rx7

**Comment by ColmMassey** [comment]
The link at the top of this issue, 

https://nextcloud.solidarityeconomy.coop/index.php/s/kQXnTxEDYFTmk8o

doesn't seem to editable via the browser.

**Comment by ColmMassey** [comment]
https://nextcloud.solidarityeconomy.coop/index.php/s/YDctt5nbP4e7Rx7 , the one I have been editing seems to point to 
Mutual-Aid-Mini.ods in the ..CodeOperatives/MutualAid folder

**Comment by ColmMassey** [comment]
NextcloudDownloadTest.csv is the one you are currently linked to in a NextCloud folder called Public

**Comment by ColmMassey** [comment]
I have overwritten NextcloudDownloadTest.csv with other files, content. What were the restrictions on using spreadsheets in Nextcloud for this purpose?

**Comment by ColmMassey** [comment]
Did you find what was stopping the map from updating? 
Just checking I am looking at the right url for the manually updated spreadsheet

https://dev.mutual-aid.solidarityeconomy.coop/ ?

**Comment by ColmMassey** [comment]
@wu-lee What is holding this back form working?

**Comment by wu-lee** [comment]
There's a problem with the mutual aid scripts, but so far I've been focusing on getting the email notification of this problem working rather than fixing the problem itself, because I'd like to see evidence that the failure notifications arrive. This has involved an email exchange with WebArchitects.

There's now a reply from WebArchitects which is saying we can use their mail system to relay the notifications, I just need to make this work (possibly with a bit of help from them).  There are other alternatives, but they tend to involve trying to use gratis-tier services on things like Mailgun, which tends to have rate limiting, or setting up our own mail server, which is time consuming and requires ongoing maintenance.

If this doesn't work out soon though, I'll just fix the problem and break it again later to test the notifications.

**Comment by ColmMassey** [comment]
I haven't been able to follow the mail server notification issues, but I think we may need to put that aside and fix the map updating problem now.

**Comment by wu-lee** [comment]
Ok, I've fixed mutual aid and the job is running ok now.

There were a couple of problems, one trivial, one a bit of a headache.

The trivial problem was just a configuration error - some settings from the config I use for testing locally had leaked in and the deployment was trying to upload to dev-0 without a password. Easy fix.

The not-so-easy problem was that the new Nextcloud spreadsheet URL does *not* work like the old one. The old one gets a CSV file, presumably because the actual file stored is a CSV.  The new one is an Open-Office .ods file, and does not download as a CSV, but a binary Open-Office blob.  I thought I'd checked this was working during our meeting last week, but I must have made a mistake.

There seems to be no option for this to be converted to CSV, so I needed to modify the config file and add a new dependency on a Ruby module which can read .ods format files ([Roo][1]). Cue other problems with [bugs in Roo][2] and installation errors.

But, these are squished now, and the conversion works satisfactorily. You can see the resulting map here:

https://dev.mutual-aid.solidarityeconomy.coop/

Of course, there may be problems to solve for that too now. For instance, they all seem to be labelled "Co-operative"... 

[1]: https://github.com/roo-rb/roo
[2]: https://github.com/roo-rb/roo/issues/545

**Comment by ColmMassey** [comment]
Nice. Great that we now have ods support.
Leave display of org type for the moment. Add description & twitter.

**Comment by wu-lee** [comment]
Regarding the mail issues, I concluded that WebArch's mail server isn't really equipped to make it easy to use it as an SMTP gateway for notifications from the sausage machine. Therefore for now, having done a bit pf RTFM I've managed to use my own mail serve rat noodlefactory.co.uk as an SMTP gateway.

This isn't ideal since it's my infrastructure, not SEA's, but works for now. Possibly a better option is to pay a few dollars per month for a [pay-as-you-grow Mailgun account][1], specifically about 80 cents per 1000 emails.  In the worst case and it sends once every ten minutes, there'll be 6*24*31 = ~4.5k emails per server, so about $9/month for both servers. But the mails should drop to zero now it's mailing only on an error.

[1]: https://www.mailgun.com/pricing/

**Comment by ColmMassey** [comment]
> Possibly a better option is to pay a few dollars per month for a [pay-as-you-grow Mailgun account](https://www.mailgun.com/pricing/),

Happy to do that under Digital Commons. No point setting up on SEA account and then have the fun of transferring.

**Comment by wu-lee** [comment]
Ok, I've filed off some rough edges to do with sea-map with things which were added for the ICA (mainly that a map site which didn't use vocabs would make it throw an error), and bumped the dependency of the `mutual-aid-project` map site to a newly released `sea-map@1.0.0` (ICA-style vocab support, ESSGLOBAL 2.1, WebPack, plus John's dialog improvements)

Also v2.1.0 of `se-open-data` - with some Twitter normalisation code.

Finally, redeployed this map using all of the above.

https://dev.mutual-aid.solidarityeconomy.coop/

Meawhile, seems that the data has been modified, and there is only one (approved) initiative, "NYC Mutual Aid". Knowing that, this seems to be working.

So overall, mostly all done (except I see back at the beginning this was intended to be a multi-source map?)
